Std/Data/Ord.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2022 Jannis Limperg. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jannis Limperg
+-/
+
+namespace Ordering
+
+/--
+Check whether the ordering is 'equal'.
+-/
+def isEQ : Ordering → Bool
+  | eq => true
+  | _ => false
+
+/--
+Check whether the ordering is 'less than'.
+-/
+def isLT : Ordering → Bool
+  | lt => true
+  | _ => false
+
+/--
+Check whether the ordering is 'greater than'.
+-/
+def isGT : Ordering → Bool
+  | gt => true
+  | _ => false
+
+/--
+Check whether the ordering is 'greater than or equal'.
+-/
+def isGE : Ordering → Bool
+  | lt => false
+  | _ => true
+
+end Ordering
+
+/--
+Compare `x` and `y` lexicographically by `cmp₁` and `cmp₂`. `x` and `y` are
+first compared by `cmp₁`. If this returns 'equal', `x` and `y` are compared
+by `cmp₂` to break the tie.
+-/
+def compareLex (cmp₁ cmp₂ : α → α → Ordering) (x y : α) : Ordering :=
+  match cmp₁ x y with
+  | .eq => cmp₂ x y
+  | ord => ord
+
+/--
+Compare `x` and `y` by comparing `f x` and `f y`.
+-/
+def compareOn [ord : Ord β] (f : α → β) (x y : α) : Ordering :=
+  compare (f x) (f y)
+
+
+namespace Ord
+
+/--
+Derive a `BEq` instance from an `Ord` instance.
+-/
+def toBEq (ord : Ord α) : BEq α where
```

Comment:
All of the functions here need to be `protected`.

Reply:
Done. What's the general rule wrt `protected`?